src/Form/Type/SecurityRolesType.php

```diff
@@ -135,7 +137,7 @@ public function configureOptions(OptionsResolver $resolver): void
         ]);
 
         // Symfony 2.8 BC
-        if ($resolver->isDefined('choices_as_values')) {
+        if (method_exists(FormTypeInterface::class, 'setDefaultOptions')) {
```

Comment:
Can you explain the fix?

Reply:
choices_as_values is removed in Symfony 4

Reply:
Which means `$resolver->isDefined('choices_as_values')` will return `false`, which means `$resolver->setDefault('choices_as_values', true);` will not be set to `true`, which… makes sense if that option is removed?

Reply:
The default value of `choices_as_values` is false and that shows deprecation notices, but we do not have that option in SF4, so he checks if it is < SF3 and sets it to true to avoid deprecation notices. Am I correct @jordisala1991 ?

Reply:
Yes, that's also probably. I just wanted to remove the deprecations on Sf3, but wasn't sure if this would work on Sf4